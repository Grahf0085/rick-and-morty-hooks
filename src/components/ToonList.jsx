import React from 'react';
import PropTypes from 'prop-types';
import Toon from './Toon';
import './toonlist.css';

const ToonList = ({ toons }) => {
  const toonElements = toons.map((toon) => (
    <li key={toon.id}>
      <Toon {...toon} />
    </li>
  ));

  return <ul>{toonElements}</ul>;
};

ToonList.propTypes = {
  toons: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string,
      image: PropTypes.string.isRequired
    })
  ).isRequired,
};

export default ToonList;
