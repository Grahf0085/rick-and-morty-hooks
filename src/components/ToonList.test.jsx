import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter as Router } from 'react-router';
import ToonList from './ToonList';

describe('ToonList component', () => {
  it('makes ul filled with lis from Toon component', () => {
    const toons = [{
      'id': '1',
      'image': 'https://rickandmortyapi.com/api/character/avatar/1.jpeg'
    }];

    const { asFragment } = render(
      <Router>
        <ToonList toons={toons} />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
