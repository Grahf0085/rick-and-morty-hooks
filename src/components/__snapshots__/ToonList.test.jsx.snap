// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToonList component makes ul filled with lis from Toon component 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <figure>
        <a
          href="/character/1"
        >
          <img
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          />
        </a>
      </figure>
    </li>
  </ul>
</DocumentFragment>
`;
